astroquery/utils/tap/core.py

```diff
@@ -1648,6 +1652,131 @@ def delete_user_table(self, table_name=None, force_removal=False,
+            dict with pairs "old_column1_name:new_column1_name"
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+
+        Example
+        -------
+        rename_table(table_name=old_table_name, new_table_name=new_table_name,
+        new_column_names_dict=[old_column1:new_column1, old_column2:new_column2, ...])
+        """
+        if new_column_names_dict is None:
+            new_column_names_dict = {}
+        args = {}
+
+        if table_name is None:
+            raise ValueError("Table name cannot be null")
+        if (new_table_name is None or new_table_name == '') and \
+                (new_column_names_dict is None or not new_column_names_dict):
+            raise ValueError("Please introduce as minimum a new table tame or a new name for a column with format "
+                             "old_column1_name:new_column1_name, ... ,old_columnN_name:new_columnN_name")
+        if new_table_name is not None or new_table_name != '':
+            if new_column_names_dict is None or not new_column_names_dict:
+                # case 1: We only need to rename the table
+                args = self.get_args_4_rename_table_only_table_name(table_name, new_table_name)
+            else:
+                # case 2: We need to rename both, columns and column name
+                args = self.get_args_4_rename_table_all(table_name, new_table_name, new_column_names_dict)
+
+        if new_table_name is None or new_table_name == '':
+            if new_column_names_dict is not None or new_column_names_dict:
+                # case 3: We only need to rename the columns but same column name
+                args = self.get_args_4_rename_table_only_columns(table_name, new_column_names_dict)
+
+        connHandler = self.__getconnhandler()
+        data = connHandler.url_encode(args)
+        response = connHandler.execute_table_tool(data, verbose=verbose)
+        if verbose:
+            print(response.status, response.reason)
+            print(response.getheaders())
+        connHandler.check_launch_response_status(response,
+                                                 verbose,
+                                                 200)
+        msg = f"Table '{table_name}' updated."
+        print(msg)
+
+    def get_args_4_rename_table_only_table_name(self, table_name, new_table_name):
+
+        args = {
+            "action": "rename",
+            "new_table_name": new_table_name,
+            "table_name": table_name
+        }
+        return args
+
+    def get_args_4_rename_table_all(self, table_name, new_table_name, new_column_names_dict):
+        count = 0
+        new_column_names = ""
+        # check if the changes proposed for the columns are correct.
+        for key, value in new_column_names_dict.items():
+            if key == '':
+                raise ValueError(f" Old column name introduced "f"{key}"f" cannot be empty")

```

Comment:
Instead of checking if `key == ''` wouldn't it be better to check if `key` is actually present in the table? Or is that not possible?